packages/shared/src/components/post/write/MultipleSourceSelect.tsx

```diff
@@ -0,0 +1,294 @@
+import {
+  SourceMemberRole,
+  SourceType,
+  SourcePermissions,
+} from '../../../graphql/sources';
+import {
+  ButtonSize,
+  ButtonVariant,
+  ButtonIconPosition,
+} from '../../buttons/common';
+import { ProfileImageSize } from '../../ProfilePicture';
+import { cloudinarySquadsImageFallback } from '../../../lib/image';
+
+import { Button } from '../../buttons/Button';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../../typography/Typography';
+import { useAuthContext } from '../../../contexts/AuthContext';
+import { TruncateText } from '../../utilities';
+import type { LoggedUser } from '../../../lib/user';
+import { PopoverFormContainer } from '../../../features/common/components/PopoverFormContainer';
+import type { CheckboxProps } from '../../fields/Checkbox';
+import { Checkbox } from '../../fields/Checkbox';
+import { IconSize } from '../../Icon';
+import { verifyPermission } from '../../../graphql/squads';
+import SpamWarning from '../../widgets/SpamWarning';
+import { Tooltip } from '../../tooltip/Tooltip';
+import ConditionalWrapper from '../../ConditionalWrapper';
+import { anchorDefaultRel } from '../../../lib/strings';
+import { webappUrl } from '../../../lib/constants';
+import { LazyImage } from '../../LazyImage';
+
+const defaultSquad = {
+  image: cloudinarySquadsImageFallback,
+  permalink: null,
+  active: true,
+  public: false,
+  membersCount: 1,
+  description: null,
+  memberPostingRole: SourceMemberRole.Admin,
+  memberInviteRole: SourceMemberRole.Admin,
+};
+
+export const generateDefaultSquad = (username: string): Squad => ({
+  ...defaultSquad,
+  id: username,
+  handle: username,
+  name: `${username}'s public Squad`,
+  type: SourceType.Squad,
+  memberPostingRole: SourceMemberRole.Moderator,
+  memberInviteRole: SourceMemberRole.Member,
+  moderationPostCount: 0,
+  moderationRequired: false,
+});
+
+export const generateUserSourceAsSquad = (user: LoggedUser): Squad => ({
+  ...generateDefaultSquad(user.username),
+  id: user.id,
+  name: user.name,
+  handle: user.id,
+  // @ts-expect-error Intentionally using a UserSource as a Squad
+  type: SourceType.User,
+  image: user.image,
+});
+
+const Label = ({ children }: PropsWithChildren) => (
+  <Typography type={TypographyType.Caption1} color={TypographyColor.Quaternary}>
+    {children}
+  </Typography>
+);
+
+const CreateNewSquadLink = () => (
+  <a
+    className="flex items-center gap-2"
+    href={`${webappUrl}squads/new`}
+    rel={anchorDefaultRel}
+    target="_blank"
+  >
+    <LazyImage
+      imgSrc={defaultSquad.image}
+      imgAlt="New squad image"
+      className="size-8 rounded-full"
+      aria-hidden
+      role="presentation"
+    />
+    <Typography
+      bold
+      className="flex-1"
+      color={TypographyColor.Primary}
+      type={TypographyType.Callout}
+    >
+      Create a new squad
+    </Typography>
+    <ArrowIcon aria-hidden className="rotate-90" />
+  </a>
+);
```

Comment:
We are opening the squad creation page in a new tab. 
https://dailydotdev.slack.com/archives/C02AA18RD1D/p1759045580468009?thread_ts=1758876007.441899&cid=C02AA18RD1D